packages/type-storage/src/utils/createFunction.ts

```diff
@@ -59,13 +60,19 @@ export default function createFunction (
 
       const type = meta.type.asMap.key.toString(); // Argument type, as string
 
-      return xxhash(
+      const methodAndArgs = xxhash(
         u8aConcat(
           u8aFromUtf8(`${section.toString()} ${method.toString()}`),
           createType(type, arg).toU8a(true)
         ),
         128
       );
+
+      // StorageKey is a Bytes, so is length-prefixed
+      return u8aConcat(
+        Compact.encodeU8a(methodAndArgs.length, DEFAULT_LENGTH_BITS),
```

Comment:
As discussed, not sure about this at all. If I look at what works, it is just the hex, with no length -

`freeBalance('5GoKvZWG5ZPYL1WUovuHW3zJBWBP5eT8CbqjdRY4Q6iMaDtZ')` -> `0xce3f3d8f09e3411403f5ca59d042a40e` 16 bytes, no length prefix. (That works as expected on the UI)

Reply:
... Then again, once passed to storageKey it will whack the length.